java/servers/src/org/xtreemfs/common/libxtreemfs/jni/NativeHelper.java

```diff
@@ -0,0 +1,215 @@
+/*
+ * Copyright (c) 2015 by Johannes Dillmann, Zuse Institute Berlin
+ *
+ * Licensed under the BSD License, see LICENSE file for details.
+ *
+ */
+package org.xtreemfs.common.libxtreemfs.jni;
+
+import java.io.File;
+import java.math.BigInteger;
+import java.net.URL;
+import java.util.Arrays;
+import java.util.List;
+
+import org.xtreemfs.common.libxtreemfs.Options;
+import org.xtreemfs.common.libxtreemfs.jni.generated.ClientProxy;
+import org.xtreemfs.common.libxtreemfs.jni.generated.IntVector;
+import org.xtreemfs.common.libxtreemfs.jni.generated.OptionsProxy;
+import org.xtreemfs.common.libxtreemfs.jni.generated.SSLContext;
+import org.xtreemfs.common.libxtreemfs.jni.generated.SSLOptionsProxy;
+import org.xtreemfs.common.libxtreemfs.jni.generated.ServiceAddresses;
+import org.xtreemfs.common.libxtreemfs.jni.generated.StringMap;
+import org.xtreemfs.common.libxtreemfs.jni.generated.StringVector;
+import org.xtreemfs.foundation.SSLOptions;
+import org.xtreemfs.foundation.pbrpc.generatedinterfaces.RPC.UserCredentials;
+import org.xtreemfs.pbrpc.generatedinterfaces.GlobalTypes.KeyValuePair;
+
+public final class NativeHelper {
+
+    /**
+     * Load the library with the platform independent name. (f.ex. jni-xtreemfs instead of libjni-xtreemfs.so) <br>
+     * Locally built libraries from the source tree are preferred. On Linux feasible directories within the FHS are
+     * searched. Finally the common library path is searched.
+     * 
+     * @param name
+     */
+    public static void loadLibrary(String name) {
```

Comment:
We should ensure that this works also for Hadoop. Placing the library in the Hadoop lib dir might be useful to be able to install Hadoop without root permissions.